src/App.tsx

```diff
@@ -1,61 +1,118 @@
 import './App.scss';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUser(UserId: number) {
+  const foundUser = usersFromServer.find(user => user.id === UserId);
+
+  return foundUser || null;
+}
+
+const todos: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [userId, setUserId] = useState(0);
+  const [userIdError, setUserIdError] = useState(false);
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+
+  const handleTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setTitleError(false);
+    setTitle(value);
+  };
+
+  const handleuserid = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    setUserIdError(false);
+    setUserId(Number(value));
+  };
+
+  const handleSubmitForm = (event: React.ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const lastId = Math.max(0, ...todos.map(({ id }) => id));
+
+    const newTodo: Todo = {
+      id: lastId + 1,
+      userId,
+      title,
+      completed: false,
+      user: getUser(userId),
+    };
+
+    if (userId && title.trim()) {
+      todos.push(newTodo);
+    } else {
+      setTitleError(title.trim() === '');
+      setUserIdError(userId === 0);
+
+      if (title.trim() === '') {
```

Comment:
```if (userId && title.trim()) {
      todos.push(newTodo);
    } else {
      setTitleError(title.trim() === '');
      setUserIdError(userId === 0);

      if (title.trim() === '') {
        setTitle('');
      }

      return;
    }```

Reply:
nested conditions "if" is not the best solution. Try to rewrite so that inside the "if" block you don't have another "if" block